site/src/xServices/createWorkspace/createWorkspaceXService.ts

```diff
@@ -164,21 +117,20 @@ export const createWorkspaceMachine = createMachine(
       },
     },
     guards: {
-      hasValidPreSelectedTemplate: (ctx, event) => {
-        if (!ctx.preSelectedTemplateName) {
-          return false
-        }
-        const template = event.data.find((template) => template.name === ctx.preSelectedTemplateName)
-        return !!template
-      },
       areTemplatesEmpty: (_, event) => event.data.length === 0,
     },
     actions: {
       assignTemplates: assign({
         templates: (_, event) => event.data,
       }),
       assignSelectedTemplate: assign({
-        selectedTemplate: (_, event) => event.template,
+        selectedTemplate: (ctx, event) => {
+          for (const template of event.data) {
+            if (template.name === ctx.templateName) {
+              return template
+            }
```

Comment:
Can we use Array's [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) here?

Reply:
yes, for loops are gonna die hard for me lol